### DISCUSSION TITLE
How to create local themes for styling

### DISCUSSION THREAD

**Discussion by scmay** [plantuml/plantuml#1026] [General]
I did read the page on local theme styling

I put this in my sample file

!theme qps from /path/to/themes/folder

but I am not sure where my local theme file (e.g myTheme.puml) should sit at in the file directory. The screenshot shows my current source code directory, where all the code sits in "Source" and I have put my myTheme.puml file under "themes" folder.
Is that the right spot, or should I place my file somewhere else?
![EB5AF3B2-A35C-4362-B150-F49FFE290630](https://user-images.githubusercontent.com/22205985/172546151-0763089f-b403-4544-b64f-62561da2c162.jpeg)

**Comment by AlasdairDougall** [comment]
Hi,

Just started using PlantUML in QCS, but in Queensland Corrective Services, and what would be the chance I came across a fellow officer doing the same, be it in a different country?

Just though we would sat Hi, and our little team of architects, which means two of us, will hopefully have something worth sharing in these discussions.  

All the bet,

Alasdair

**Reply by scmay** [reply]
Sorry I didn’t think it would be so obvious. Currently doing some work for QPS and have no luck with local theme colors

**Comment by AlasdairDougall** [comment]
OFFICIAL

Hi,

We are trying to create a unique theme for QCS, as the tool is powerful documentation tool, allowing a consistent approach to diagram creation.  We are working on a similar issue around styling the legend cells, so will let you know how it goes.

What issue were you trying to resolve?

Alasdair


________________________________

**Comment by scmay** [comment]
Same same
Custom color for QPS
None of the existing themes suit our organisation.

**Comment by The-Lum** [comment]
Hello @scmay, @AlasdairDougall, and all,

If you invoke theme styling with:
```puml
!theme qps from /path/to/themes/folder
```
You must rename you local theme file as `puml-theme-qps.puml`, according to `puml-theme-<theme_name>.puml`.
And putting them on the `/path/to/themes/folder` folder.

But for your case:
- Perhaps put the `puml-theme-qps.puml` file on the `Source/themes` folder;
- Then invoke, on all yours codes (on `Source` folder), by:
```puml
!theme qps from /themes
```

See doc. here:
- Constraint for naming theme: `puml-theme-<theme_name>.puml`
- https://plantuml.com/en/theme#14ed5f7d5f35bab9
	
See also similar question here:
- [QA-16079](https://forum.plantuml.net/16079/how-to-use-local-theme-colors)

_If that can help,_
Regards.

**Comment by scmay** [comment]
thanks! I got it to work. 

my path is 
!theme qps from ../themes

**Comment by AlasdairDougall** [comment]
OFFICIAL

Thanks, really helpful pointer.
________________________________
